codegenerator/cli/src/config_parsing/entity_parsing.rs

```diff
@@ -524,6 +524,38 @@ impl Entity {
     }
 }
 
+///  used to get the positive integers in the directives from the GraphQL schema.
+fn get_positive_integer(
+    arg_value: &Value<String>,
+    field_name: &str,
+    param_name: &str,
```

Comment:
My suggestion for the error pattern is don't pass these context values in here for errors. Rather make it a general parsing fn and add `.context("Parsing precision")` for eg.

Reply:
Cool, makes sense.